modules/dhcp/dhcp_api.rb

```diff
@@ -10,13 +10,15 @@ class Proxy::DhcpApi < ::Sinatra::Base
 
   get "/?" do
     content_type :json
+    server.validate_supported_address(params[:ip])
     server.subnets.map { |s| {:network => s.network, :netmask => s.netmask, :options => s.options} }.to_json
   rescue => e
     log_halt 400, e
   end
 
   get "/:network" do
     content_type :json
+    server.validate_supported_address(params[:ip])
```

Comment:
Have you tested this? :-)

```suggestion
    server.validate_supported_address(params[:network])
```